### ISSUE TITLE
npm 3.3.8 and Browserified modules

### ISSUE THREAD

**Issue by acoll** [substack/node-browserify#1409]
I have a module with transforms defined in the package.json so that when I require this module elsewhere the necessary transforms get applied. This has broken since I updated to npm 3.3.8. The problem is that when the new package.json file is written after the npm install, the browserify transforms are no longer in the correct order. For example I use
```json
"browserify": {
   "transform": ["jadeify", "babelify", "brfs"]
}
```
but after installed it is
```json
"browserify": {
   "transform": ["babelify", "brfs", "jadeify"]
}
```

Which doesn't work because babel can't parse the jade files. It looks like the problem originates from here: https://github.com/npm/npm/blob/master/lib/utils/deep-sort-object.js#L6 and https://github.com/npm/npm/blob/master/lib/install/update-package-json.js#L14

I opened an issue with npm as well but thought you guys might want to know about this since it breaks a sort of important feature of browserify.

Issue with npm: https://github.com/npm/npm/issues/10063

**Comment by mreinstein** [comment]
+1 this is breaking browserify bundling all over the place.

**Comment by substack** [comment]
Wow this seems like a big bug on npm's part if it's resorting arrays.

**Comment by substack** [comment]
PR sent upstream: https://github.com/npm/npm/pull/10185

**Comment by yaquawa** [comment]
Same problem here... 
Why don't just specify a config file just like babel ( `.babelrc` ) or webpack(webpack.config.js) does!? 
It's way better than just throws all the configurations into the package.json......

**Comment by mreinstein** [comment]
> So, to use browserify WHY you have to have a package.json? This make no sense for me.

This issue is related to a problem with transforms. It's not a place to gripe about browserify architecture.

**Comment by substack** [comment]
This bug is fixed upstream in npm 3.7.0.
